sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/DataMovementSharesExtensions.cs

```diff
@@ -235,6 +227,30 @@ public static FileSmbProperties GetFileSmbProperties(
                    : default;
         }
 
+        private static DateTimeOffset? GetChangedOnTime(
+            ShareFileStorageResourceOptions options,
+            IDictionary<string, object> sourceRawProperties)
+        {
+            if (options?._isFileChangedOnSet == true)
+            {
+                return options?.FileChangedOn;
+            }
+
+            ShareProtocol sourceProtocol = sourceRawProperties?.TryGetValue(DataMovementConstants.ResourceProperties.ShareProtocol, out object protocol) == true
```

Comment:
The protocol should be in the properties by now, right? And if not, we'd have a problem elsewhere. Perhaps we should throw if its not there rather than defaulting... not sure.

Reply:
Discussed offline, probably fine to leave as is.